azure-kusto-data/source/client.js

```diff
@@ -31,33 +31,33 @@ module.exports = class KustoClient {
         };
     }
 
-    execute(db, query, callback, properties) {
+    async execute(db, query, properties) {
         query = query.trim();
         if (query.startsWith(".")) {
-            return this.executeMgmt(db, query, callback, properties);
+            return this.executeMgmt(db, query, properties);
         }
 
-        return this.executeQuery(db, query, callback, properties);
+        return this.executeQuery(db, query, properties);
     }
 
-    executeQuery(db, query, callback, properties) {
-        return this._execute(this.endpoints.query, db, query, null, callback, properties);
+    async executeQuery(db, query, properties) {
+        return this._execute(this.endpoints.query, db, query, null, properties);
     }
 
-    executeMgmt(db, query, callback, properties) {
-        return this._execute(this.endpoints.mgmt, db, query, null, callback, properties);
+    async executeMgmt(db, query, properties) {
+        return this._execute(this.endpoints.mgmt, db, query, null, properties);
     }
 
-    executeStreamingIngest(db, table, stream, streamFormat, callback, properties, mappingName) {
+    async executeStreamingIngest(db, table, stream, streamFormat, mappingName) {
         let endpoint = `${this.endpoints.ingest}/${db}/${table}?streamFormat=${streamFormat}`;
         if (mappingName != null) {
             endpoint += `&mappingName=${mappingName}`;
         }
 
-        return this._execute(endpoint, db, null, stream, callback, properties);
+        return this._execute(endpoint, db, null, stream, null);
     }
 
-    _execute(endpoint, db, query, stream, callback, properties) {
+    async _execute(endpoint, db, query, stream, properties) {
         let headers = {};
         Object.assign(headers, this.headers);
```

Comment:
I'm not sure why we are doing it. But to prevent changing the member "headers" consider marking it as const